robot/main.py

```diff
@@ -17,4 +17,27 @@
 
 
 # Write your program here.
-ev3.speaker.beep()
+#ev3.speaker.beep()
+left_motor = Motor(Port.A)
+right_motor = Motor(Port.B)
+obstacle_sensor = UltrasonicSensor(Port.S4)
+
+#motor.run_time(360, 10000, then=Stop.HOLD, wait=True)
+
+robot = DriveBase(left_motor, right_motor, wheel_diameter=50, axle_track=90)
+
+while True:
+    # Begin driving forward at 200 millimeters per second.
+    robot.drive(200, 0)
+
+    # Wait until an obstacle is detected. This is done by repeatedly
+    # doing nothing (waiting for 10 milliseconds) while the measured
+    # distance is still greater than 300 mm.
+    while obstacle_sensor.distance() > 300:
```

Comment:
Move this while loop into a new function "check_for_wall" or "obstacle_ahead" or something like that.